routes/main.js

```diff
@@ -0,0 +1,67 @@
+const varson = require('varson');
+const aug = require('aug');
+
+const deployLog = {};
+
+exports.deploy = {
+  method: 'get',
+  path: '/{path*}',
+  handler(request, h) {
+    const server = request.server;
+
+    const host = request.headers.host.split('.')[0];
+    const defaultService = request.server.settings.app.service;
+    const hostData = request.server.settings.app.hosts;
+
+    const deploy = async function(obj) {
+      const deployKey = `${obj.name}_${obj.branch}`;
+      if (deployLog[deployKey]) {
+        const now = new Date().getTime() - (5 * 60 * 1000);
+        if (now < deployLog[deployKey]) {
+          server.log(['docker-autostart', 'info'], `${obj.name}/${obj.branch} already deploying.`);
+          return `Already deploying ${obj.name}/${obj.branch}`;
+        }
+      }
+      deployLog[deployKey] = new Date().getTime();
+      try {
+        const result = await server.req.post(obj.endpoint, { payload: obj.payload });
+        return result;
+      } catch (e) {
+        server.log(['docker-autostart', 'endpoint', 'error'], { error: e });
+        // reset deployLog
+        deployLog[deployKey] = 0;
+        return e;
+      }
+    };
+
+    let match = false;
+
+    Object.keys(hostData).forEach(repo => {
```

Comment:
I think doing a filter here and returning the repo would be better, too much logic inside the closure